### PR TITLE
Add ESLint

### PR DESCRIPTION
**Pull request by wwlorey**

### CODE REVIEW

File: src/codeFlowLogin.ts [hunk 1]
```diff
@@ -169,11 +173,14 @@ export async function login(clientId: string, environment: Environment, adfs: bo
...
 		await openUri(`http://localhost:${port}/signin?nonce=${encodeURIComponent(nonce)}`);
+		// eslint-disable-next-line @typescript-eslint/no-misused-promises
 		const redirectTimer = setTimeout(() => redirectTimeout().catch(console.error), 10*1000);
 
 		const redirectReq = await redirectPromise;
 		if ('err' in redirectReq) {
+			// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
 			const { err, res } = redirectReq;
+			// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
 			res.writeHead(302, { Location: `/?error=${encodeURIComponent(err && err.message || 'Unkown error')}` });
```
Comment (nturinski) [inline src/codeFlowLogin.ts:184]:
```suggestion
			res.writeHead(302, { Location: `/?error=${encodeURIComponent(err && err.message || 'Unknown error')}` });
```

File: src/codeFlowLogin.ts [hunk 2]
```diff
@@ -201,6 +208,7 @@ export async function login(clientId: string, environment: Environment, adfs: bo
 			res.end();
 			return tokenResponse;
 		} catch (err) {
+			// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
 			res.writeHead(302, { Location: `/?error=${encodeURIComponent(err && err.message || 'Unkown error')}` });
```
Comment (nturinski) [inline src/codeFlowLogin.ts:212]:
```suggestion
			res.writeHead(302, { Location: `/?error=${encodeURIComponent(err && err.message || 'Unknown error')}` });
```

File: tsconfig.json [hunk 3]
```diff
@@ -13,7 +13,8 @@
 			"src/typings"
 		],
 		"sourceMap": true,
-		"rootDir": "src"
+		"rootDir": ".",
+		"resolveJsonModule": true
```
Comment (nturinski) [inline tsconfig.json:17]: Why did this have to be added?  I don't see it in our extensions.

**Review by nturinski — APPROVED** [review]

File: tsconfig.json (hunk 3 above)
Comment (wwlorey) [reply tsconfig.json:17]: That's required for importing from a package.json here: https://github.com/microsoft/vscode-azure-account/blob/82554472a1866d3494b60e5ffeefd929a30c715d/src/telemetry.ts#L7

Typescript throws this error without it:
```
TS2732: Cannot find module '../package.json'. Consider using '--resolveJsonModule' to import module with '.json' extension
```

Comment (nturinski) [reply tsconfig.json:17]: So odd

**Pull request merged**